backend/de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/impexp/BankStatementImportProcess.java

```diff
@@ -175,62 +180,147 @@ private void createBankStatementLine(final I_I_BankStatement importRecord)
-				// TODO add documentNo to import
-				.statementDate(TimeUtil.asLocalDate(importBankStatement.getStatementDate()))
-				.name(importBankStatement.getName())
-				.description(importBankStatement.getDescription())
-				.eft(BankStatementCreateRequest.ElectronicFundsTransfer.builder()
-						.statementDate(TimeUtil.asLocalDate(importBankStatement.getEftStatementDate()))
-						.statementReference(importBankStatement.getEftStatementReference())
-						.build())
-				.build());
+															.orgId(OrgId.ofRepoId(importBankStatement.getAD_Org_ID()))
+															.orgBankAccountId(BankAccountId.ofRepoId(importBankStatement.getC_BP_BankAccount_ID()))
+															// TODO add documentNo to import
+															.statementDate(TimeUtil.asLocalDate(importBankStatement.getStatementDate()))
+															.name(importBankStatement.getName())
+															.description(importBankStatement.getDescription())
+															.eft(BankStatementCreateRequest.ElectronicFundsTransfer.builder()
+																		 .statementDate(TimeUtil.asLocalDate(importBankStatement.getEftStatementDate()))
+																		 .statementReference(importBankStatement.getEftStatementReference())
+																		 .build())
+															.build());
+	}
+
+	@VisibleForTesting
+	static void computeUpdateLineAmts(@NonNull final I_I_BankStatement importRecord)
+	{
+		final String amtFormat = getAmtFormat(importRecord);
+
+		if (X_I_BankStatement.AMTFORMAT_AmountPlusIndicator.equals(amtFormat))
+		{
+			final BigDecimal trxAmt;
+			final boolean isDebit = "S".equals(importRecord.getDebitOrCreditIndicator());
+			if (isDebit)
+			{
+				trxAmt = importRecord.getDebitOrCreditAmt().negate();
+			}
+			else
+			{
+				trxAmt = importRecord.getDebitOrCreditAmt();
+			}
+
+			importRecord.setTrxAmt(trxAmt);
+			final BigDecimal stmtAmt = trxAmt.add(importRecord.getChargeAmt()).add(importRecord.getInterestAmt());
+			importRecord.setStmtAmt(stmtAmt);
+		}
+		else if (X_I_BankStatement.AMTFORMAT_DebitPlusCredit.equals(amtFormat))
+		{
+			final BigDecimal stmtAmt = importRecord.getCreditStmtAmt().subtract(importRecord.getDebitStmtAmt());
+			importRecord.setStmtAmt(stmtAmt);
+			final BigDecimal trxAmt = stmtAmt.subtract(importRecord.getInterestAmt()).subtract(importRecord.getChargeAmt());
+			importRecord.setTrxAmt(trxAmt);
+		}
+		else if (X_I_BankStatement.AMTFORMAT_Straight.equals(amtFormat))
+		{
+			importRecord.setStmtAmt(importRecord.getTrxAmt());
+		}
+
+		{
+			final BigDecimal sum = importRecord.getTrxAmt().add(importRecord.getInterestAmt()).add(importRecord.getChargeAmt());
+			if (importRecord.getStmtAmt().compareTo(sum) != 0)
+			{
+				throw new AdempiereException("Invalid amount");
+			}
+		}
+	}
+
+	private static void updateDescription(@NonNull final I_I_BankStatement importRecord)
+	{
+		final String lineDescription = Joiner.on(" / ")
+				.skipNulls()
+				.join(importRecord.getLineDescription(),
```

Comment:
use de.metas.util.StringUtils#trimBlankToNull on of of the list descriptions parts.

i.e.
.join(
StringUtils.trimBlankToNull(importRecord.getLineDescription()),
StringUtils.trimBlankToNull(importRecord.getLineDescriptionExtra_1()),
....
);

Reply:
done